### PR TITLE
Corrected host/group vars

### PR DESCRIPTION
**Pull request by retr0h**
* Documented host/group var native inventory testing
   Ansible will load the directory host/group_vars if it exists
   in the root of the project.
* Added group/host vars functional scenario.
* Removed group vars symlink code.  We didn't use this.

### CODE REVIEW

**Comment by retr0h** [comment]
```
  syntax: commands succeeded
  py27-ansible19-unit: commands succeeded
  py27-ansible19-functional: commands succeeded
  py27-ansible20-unit: commands succeeded
  py27-ansible20-functional: commands succeeded
  py27-ansible21-unit: commands succeeded
  py27-ansible21-functional: commands succeeded
  doc: commands succeeded
  congratulations :)
```

File: test/functional/driver/docker/scenarios/group_host_vars/molecule.yml [hunk 1]
```diff
@@ -1,4 +1,12 @@
 ---
+ansible:
+  group_vars:
+    example:
+      - group_host_vars_group_molecule_yml: foo
+  host_vars:
+    group-vars-scenario:
+      - group_host_vars_host_molecule_yml: bar
+
```
Comment (retr0h) [inline test/functional/driver/docker/scenarios/group_host_vars/molecule.yml]: This doesn't appear to work.  Not sure why.  Do you have any idea @kireledan?

Doesn't seem that the host and group vars are getting created.

**Comment by rgreinho** [comment]
LGTM :+1:

**Comment by retr0h** [comment]
```
  syntax: commands succeeded
  py27-ansible19-unit: commands succeeded
  py27-ansible19-functional: commands succeeded
  py27-ansible20-unit: commands succeeded
  py27-ansible20-functional: commands succeeded
  py27-ansible21-unit: commands succeeded
  py27-ansible21-functional: commands succeeded
  doc: commands succeeded
  congratulations :)
```

**Pull request merged**